spec/facility_spec.rb

```diff
@@ -14,13 +29,92 @@
     end
   end
 
-  describe '#add service' do
+
+
+  describe 'add_service' do
     it 'can add available services' do
       expect(@facility.services).to eq([])
       @facility.add_service('New Drivers License')
       @facility.add_service('Renew Drivers License')
       @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+      @facility.add_service('Written Test')
+      @facility.add_service ('Renew License')
+      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration','Written Test','Renew License'])
+    end
+  end
+
+
+
+  describe 'register vehicle' do
+    it 'registers a vehicle' do
+      @facility.register_vehicle(@cruz)
+      expect(@facility.registered_vehicles).to include(@cruz)
+    end
+  end
+
+
+  describe 'registration date' do
+    it 'sets the registration date to today' do
+      @facility.register_vehicle(@cruz)
+      expect(@cruz.registration_date).to eq(Date.today)
+    end
+  end
+
+
+  describe 'collected_fees' do
+    it 'returns $25 for antique vehicles (25 years old or older)' do
+      @facility.register_vehicle(@camaro)
+      expect(@facility.collected_fees).to eq(25)
+    end
+
+    it 'returns $200 for electric vehicles' do
+      @facility.register_vehicle(@bolt)
+      expect(@facility.collected_fees).to eq(200)
+    end
+
+    it 'returns $100 for all other vehicles' do
+      @facility.register_vehicle(@cruz)
+      expect(@facility.collected_fees).to eq(100)
+    end
+  end
+
+
+  describe 'administer_written_test'do
+      it 'decides if the written test will be taken' do
+      expect(@facility.administer_written_test(@registrant_1)).to eq(true)
+      expect(@registrant_1.license_data[:written]).to eq(true)
+    end
```

Comment:
This method (and the following two methods) has 3 conditions that must be met in order for the return value to be true. That means we need to test all the different scenarios to ensure that we are always getting the correct return value. So we should be testing that if the facility doesn't have the service, it should return false, even if the registrant is 16 and they have their permit. But if the facility does have the service and the registrant is 13, then this should return false, etc.
When testing, we want to think about all possible return values and test for all of those.